### PR TITLE
Fix ByteString.empty comparison problem

### PR DESCRIPTION
**Pull request by gdziadkiewicz**
While working on reenabling ignored property tests for monoids based on `ByteString` defined in `FSharpx.Extras` I stumbled upon this bug, `ByteString` being marked as struct results in it getting a "free" parameterless constructor that creates it with `(null, 0, 0`) (defaults for `byte[]` and `int`). This constructor is later used to implement `ByteString.empty`. The problem is that it does not play well with comparison and equality (it is based on comparison).

I added test cases that illustrate the issue. Solution finding in progress.

### CODE REVIEW

File: tests/FSharpx.Collections.Tests/ByteStringTest.fs [hunk 1]
```diff
@@ -13,6 +13,10 @@ module ByteStringTests =
           "empty1",   ByteString.create ""B,   ByteString.create ""B,    0
           "empty2",   ByteString.create ""B,   ByteString.create "a"B,  -1  
           "empty3",   ByteString.create "a"B,  ByteString.create ""B,    1
+          "empty4",   ByteString.empty,        ByteString.empty,         0
+          "empty5",   ByteString.empty,        ByteString.create "a"B,  -1  
+          "empty6",   ByteString.create "a"B,  ByteString.empty,         1
+          "empty7",   ByteString.empty,        ByteString.create ""B,    0
```
Comment (gdziadkiewicz) [inline tests/FSharpx.Collections.Tests/ByteStringTest.fs:19]: This results in `NullReferenceException`:
```
System.NullReferenceException: Object reference not set to an instance of an object.
   at FSharpx.Collections.ByteString.Compare(ByteString a, ByteString b) in C:\Users\grzeg\repos\FSharpx.Collections\src\FSharpx.Collections\ByteString.fs:line 35
```

**Comment by gdziadkiewicz** [comment]
The tests are not being executed, see: https://github.com/fsprojects/FSharpx.Collections/actions/runs/2998970505/jobs/4812175674#step:7:307

**Comment by gdziadkiewicz** [comment]
> The tests are not being executed, see: https://github.com/fsprojects/FSharpx.Collections/actions/runs/2998970505/jobs/4812175674#step:7:307

Fixing that in #192

File: tests/FSharpx.Collections.Tests/ByteStringTest.fs (hunk 1 above)
Comment (gdziadkiewicz) [reply tests/FSharpx.Collections.Tests/ByteStringTest.fs:19]: https://github.com/gdziadkiewicz/FSharpx.Collections/actions/runs/3043148096/jobs/4902097301#step:7:491

**Review by panesofglass — APPROVED** [review]

**Pull request merged**